apps/daffio/src/app/content/docs/api-docs/containers/api-doc/api-doc.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+import { DaffioDocService } from '../../services/api-doc.service';
+import { DaffioApiDoc } from '../../models/api-doc';
+
+@Component({
+  selector: 'daffio-api-doc-container',
+  templateUrl: './api-doc.component.html',
+  styleUrls: ['./api-doc.component.scss']
+})
+export class DaffioApiDocContainer implements OnInit {
+
+  /**
+   * An API document.
+   */
+  doc: DaffioApiDoc = null;
+
+  constructor(
+    private docService: DaffioDocService,
+    private route: ActivatedRoute
+  ) { }
+
+  ngOnInit() {
+    // todo should navigate to 404 on a 404 error.
+    this.route.url.subscribe((url) => {
+      this.docService.getDoc(url[0] + '/' + url[1] + '/' + url[2]).subscribe(
```

Comment:
I think this double subscribe should be replaced by a switch map. You also need to make sure you destroy the subscription, otherwise you leave a memory leak.

Reply:
replaced with switchMap, and removed the subscription with an async pipe